activesupport/test/inflector_test.rb

```diff
@@ -33,6 +33,12 @@ def test_pluralize_empty_string
     assert_equal "", ActiveSupport::Inflector.pluralize("")
   end
 
+  def test_pluralize_with_fallback
+    I18n.stub(:default_locale, :"en-GB") do
+      assert_equal "days", ActiveSupport::Inflector.pluralize("days")
```

Comment:
I'm afraid that the above test doesn't check new code introduced by the PR. Value passed to `pluralize` is already pluralized. And even if it wasn't pluralized, it still would work correctly without the new code. 

Example using Rails 6.1.3:

```
2.7.0 :001 > I18n.default_locale
 => :"en-GB" 
2.7.0 :002 > ActiveSupport::Inflector.pluralize("day")
 => "days" 
2.7.0 :003 > ActiveSupport::Inflector.pluralize("day", :'en-GB')
 => "day" 
```

Only the last line returns incorrect result, which is fixed by this PR.

Reply:
You are right @ocher, the first problem was fixed after I noticed it in a follow-up:
https://github.com/rails/rails/pull/40774

The second issue I'll have to investigate. There was previously another part of this PR and I likely made a mistake when removing it.